subprojects/core/src/main/java/org/gradle/api/internal/TaskInternal.java

```diff
@@ -126,4 +127,15 @@ public interface TaskInternal extends Task, Configurable<Task> {
      */
     @Internal
     TaskDependency getLifecycleDependencies();
+
+    @Internal
+    List<Finalizer> getFinalizers();
+
+    interface Finalizer {
+
+        TaskDependencyInternal getTaskDependencies();
+
+        void configure();
+
+    }
```

Comment:
💭 I think this should definitely be a top-level type.  I wonder if it makes sense for anything else to be a part of these?  Do they have names/descriptions, for instance?